### ISSUE TITLE
Crash in DialogFragment. ID does not reference a View inside this View

### ISSUE THREAD

**Issue by SwiftyWang** [androidbroadcast/ViewBindingPropertyDelegate#55]
I am using this library 1.4.5 version.
With this function
`public inline fun <F : Fragment, T : ViewBinding> Fragment.viewBinding(
    crossinline vbFactory: (View) -> T,
    @IdRes viewBindingRootId: Int
)`
And working in activity and fragment.
When using in DialogFragment.
The app crash when I try to use viewBinding instance in onViewCreated callback.
After some investigated, found the dialog contentView will be set in onActivityCreated which is called after onViewCreated.
But the delegate will get the view from dialog inside.

Is that a bug? Please give some suggestions, thanks.

**Comment by kirich1409** [comment]
Hi. I need to see code of the Fragment where you have the issue

**Comment by kirich1409** [comment]
And stacktrace when crash occured

**Comment by SwiftyWang** [comment]
Normally we use DialogFragment like using a normal fragment. We create view in onCreateView and use views in onViewCreated (or onCreateView).
When use _Fragment.viewBinding_ with _viewBindingRootId_ . It will throw exception in Utils L46. Seems view hasn't added into dialog window. I checked androidx DialogFragment source code. It add view to dialog in onActivityCreated which after onCreateView.

Now I am using viewBinding function without given the rootViewId seems working. Not sure if there are any drawbacks.

**Comment by outofdate** [comment]
Yeah, we got the same issue, as we are accessing bindings in `onViewCreated`. After some digging we find out the same reason - `dialog window` view is without our view, as it is added in `onActivityCreated` of `DialogFragment`.

**Comment by NZeee** [comment]
Is there any reason why the author is finding the view in `dialog view`, not `fragment view`? Have the same issue

**Comment by kirich1409** [comment]
`DialogFragment` can be used as usual Fragment that will be added as View in layout or can be shown as Dialog. A `ViewBinding` require a View. I will think how it can be done better. Right now `DialogFragment` implementation isn't good a don't work everywhere. Let's discuss how to do that better

**Comment by rantianhua** [comment]
@kirich1409 Maybe this way is better:
```
    val view = dialog?.findViewById<View>(viewBindingRootId)
    if (view != null) {
        return view
    }
    return requireView().requireViewByIdCompat(viewBindingRootId)
```

Find the view from the dialog at first, then find it from the fragment view.
